x-pack/plugins/observability_ai_assistant/server/routes/chat/route.ts

```diff
@@ -41,6 +42,8 @@ const chatRoute = createObservabilityAIAssistantServerRoute({
 
     const client = await service.getClient({ request });
 
+    client.setChatContext(params.body.chatContext);
```

Comment:
I don't think we should do this on the server (having side-effects). I also don't think we need it for `chat`, just `/chat/complete`. Let's make it a property of `client.complete({ chatContext, ... })`

Reply:
1. Could you elaborate why this doesn't need to happen on the server when other functions do, from a conceptual standpoint? What is it about this that makes it doesn't need to happen on the server?
1. How would that work when the context function now retrieves both knowledge base entries and chat context? The kb entry getting happens on the server, so how should it then get stuff from the client here?

Reply:
> Could you elaborate why this doesn't need to happen on the server when other functions do, from a conceptual standpoint? What is it about this that makes it doesn't need to happen on the server?
>
> How would that work when the context function now retrieves both knowledge base entries and chat context? The kb entry getting happens on the server, so how should it then get stuff from the client here?

Sorry, I realize my wording is ambiguous. What I mean is that I don't think we should have side-effects on the server. The client is unique per request, but I still think it is unexpected for it to have side-effects like this. Maybe side-effects is the wrong term anyway, but what I mean is that it should be a property of /chat/complete. E.g. we don't use `setMessages` either.

> Re: also adding it to /chat: wouldn't having that also unlock augmenting the contextual insights with context?

The contextual insights use /chat/complete.